src/EppoClient.php

```diff
@@ -361,16 +377,183 @@ private function getAssignmentDetail(
+            } else {
+                throw EppoClientException::from($e);
+            }
+        }
+    }
+
+    /**
+     * @param string $flagKey
+     * @param string $subjectKey ,
+     * @param AttributeSet $subject
+     * @param array<string, AttributeSet> $actionsWithContext
+     * @param string $defaultVariation
+     * @return BanditResult
+     *
+     * @throws InvalidConfigurationException
+     * @throws HttpRequestException
+     * @throws InvalidApiKeyException
+     * @throws InvalidArgumentException
+     * @throws EppoClientException
+     * @throws BanditEvaluationException
+     */
+    private function getBanditDetail(
+        string $flagKey,
+        string $subjectKey,
+        AttributeSet $subject,
+        array $actionsWithContext,
+        string $defaultVariation
+    ): BanditResult {
+        Validator::validateNotBlank($flagKey, 'Invalid argument: flagKey cannot be blank');
+
+        $isBanditFlag = $this->configurationLoader->isBanditFlag($flagKey);
+
+        if (empty($actionsWithContext) && $isBanditFlag) {
+            // This exception is caught in graceful mode and the default is returned (@see getBanditAction)
+            throw new BanditEvaluationException("No actions provided for bandit flag {$flagKey}");
+        }
+
+        $variation = $this->getStringAssignment(
+            $flagKey,
+            $subjectKey,
+            $subject->toArray(),
+            $defaultVariation
+        );
+
+        if (!$isBanditFlag) {
+            // It's likely that the developer made a mistake passing a non-bandit flag so let's warn them.
+            syslog(LOG_WARNING, "[Eppo SDK]: Flag \"{$flagKey}\" does not contain a Bandit");
+
+            // Return the computed variation without doing any more Bandit work.
+            return new BanditResult($variation);
+        }
+
+        $banditKey = $this->configurationLoader->getBanditByVariation($flagKey, $variation);
+        if ($banditKey == null) {
+            // The assigned variation is not a bandit.
+            return new BanditResult($variation);
+        }
+
+        $bandit = $this->configurationLoader->getBandit($banditKey);
+        if ($bandit == null) {
+            throw new BanditEvaluationException(
+                "Assigned bandit not found for ($flagKey, $variation)",
+                EppoException::BANDIT_EVALUATION_FAILED_BANDIT_MODEL_NOT_PRESENT
+            );
+        }
+
+        $result = $this->banditEvaluator->evaluateBandit(
+            $flagKey,
+            $subjectKey,
+            $subject,
+            $actionsWithContext,
+            $bandit->modelData
+        );
+
+        $banditActionLog = BanditActionEvent::fromEvaluation(
+            $variation,
+            $result,
+            $bandit,
+            (new SDKData())->asArray()
+        );
+
+        if ($this->eventLogger instanceof IBanditLogger) {
+            $this->eventLogger->logBanditAction($banditActionLog);
```

Comment:
if this method doesn't have a wrapping try-catch, we should wrap it here. We don't want logging failures to tank evaluation

Reply:
done